pkg/virt-handler/options_test.go

```diff
@@ -0,0 +1,173 @@
+package virthandler
+
+import (
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	. "github.com/onsi/gomega/gstruct"
+
+	"kubevirt.io/kubevirt/pkg/virt-handler/node-labeller/api"
+)
+
+const memorySize = 16256896
+const pageAmount = 1000
+const cellDistance = 10
+const unit = "KiB"
+
+var _ = Describe("Parsing VMI Options", func() {
+	Context("virt-handler VM processes VMI options during update", func() {
+		DescribeTable("it should convert api capabilities to cmdv1 topology", func(cellCount, pageCount, cpuCount int) {
+			caps := newLibvirtCapsTopology(
+				withCells(cellCount),
+				withCellMemory(memorySize, unit),
+				withCellPages(pageCount, pageAmount, unit),
+				withCellDistances(cellDistance),
+				withCellCpus(cpuCount),
+			)
+			topology := caps.Host.Topology
+			expectedTopology := capabilitiesToTopology(caps)
+
+			Expect(expectedTopology.NumaCells).To(HaveLen(cellCount))
+			for i, cell := range topology.Cells.Cell {
+				Expect(expectedTopology.NumaCells[i].Memory).To(PointTo(matchMemory(cell.Memory)))
+				Expect(expectedTopology.NumaCells[i].Pages).To(ConsistOf(matchPages(cell.Pages)))
+				Expect(expectedTopology.NumaCells[i].Distances).To(ConsistOf(matchDistances(cell.Distances.Sibling)))
+				Expect(expectedTopology.NumaCells[i].Cpus).To(ConsistOf(matchCPUs(cell.Cpus.CPU)))
+			}
+		},
+			Entry("Single cell, single page, single CPU per cell", 1, 1, 1),
+			Entry("Multiple cells, multiple pages, multiple CPUs per cell", 4, 3, 4),
+		)
+	})
```

Comment:
I’m a bit unsure about the process of writing this test in one commit and then modifying it in another. It’s making the review a bit challenging for me.

Reply:
I understand, however, it was done to show that previous logic was functioning properly so that the new logic wouldn't seem too abrupt of an introduction. I can remove it should it be required, but the code itself doesn't change much between the two commits.

Reply:
Thank you for explaining. From a review perspective, if future commits modify code already changed in this PR, I’ll need to review it again. This makes reviewing commit-by-commit a bit challenging.